src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1325,7 +1300,8 @@ void MarkEntireAssembly (AssemblyDefinition assembly)
 			MarkCustomAttributes (assembly.MainModule, new DependencyInfo (DependencyKind.AssemblyOrModuleAttribute, assembly.MainModule), null);
 
 			if (assembly.MainModule.HasExportedTypes) {
-				// TODO: This needs more work accross all steps
+				foreach (var exportedType in assembly.MainModule.ExportedTypes)
+					_context.MarkingHelpers.MarkExportedType (exportedType, assembly.MainModule, new DependencyInfo (DependencyKind.ModuleOfExportedType, assembly.MainModule));
```

Comment:
Same feedback I left in https://github.com/mono/linker/pull/1781#discussion_r563930820:

> This looks like a significant behavior change:
> 
>- We didn't used to keep type forwards in copy assemblies - this could result in some extra facade dlls being kept that weren't needed previously. It's probably uncommon, since facades are usually going to be copyused as part of the framework, but it's unclear how copyused is supposed to behave now. Should a marked copyused assembly also have exported types kept? (This would be a bigger issue since copyused are more likely to pull in facades).
>- I _think_ this will prevent SweepStep from sweeping assembly refs from copy assemblies, so it will never update scopes for Copy assemblies. Same question - how should this work for copyused assemblies?
> 
> If we go this direction and mark all copy type forwarders, I think we can simplify a few cases in SweepStep:
> 
>- get rid of the copy case from UpdateAssembliesReferencingRemovedAssembly
>- get rid of the SweepTypeForwarders handling for copy assemblies

Reply:
To clarify, I see two options:
1. Ensure that copy *means* copy. The file is not modified, we keep exported types in it, we don't rewrite typerefs - so we need to keep referenced exported types in other assemsblies to ensure the typerefs remain valid.
2. We keep the behavior that copy has today - it may modify the file to rewrite type refs, sweep unmarked exported types in it - and later introduce an additional option (copyoriginal?) that has the behavior of 1.

Reply:
This is trying to accomplish the second option you mention. The `copy` action should behave the way it does today (mostly) -- the relevant change is we now keep the facade of marked exported types. I'm adding the relevant tests for this.